src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -162,21 +187,31 @@ function useChatUtilsHook(): UseChatUtilsHook {
       channelData.channelPicture = channelImage;
       channelData.name = channelName;
 
-      await channelData.save(localDb);
-    }
-
-    const promise = originalMembers?.map((member) => {
-      return new Promise((resolve, reject) => {
-        const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
-        try {
-          user.saveOrUpdateIfExists(localDb).then(() => resolve(null));
-        } catch (e) {
-          reject(e);
+      queue.addPriorityJob({
+        label: `saveChannelData-${channel?.id}`,
+        priority: QueueJobPriority.MEDIUM,
+        task: async () => {
+          channelData.saveIfLatest(localDb);
         }
       });
-    });
+    }
 
-    await Promise.all(promise).catch((e) => console.log('saveChannelDetail error', e));
+    originalMembers?.map((member) => {
+      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+      try {
+        queue.addPriorityJob({
+          label: `save user from channel detail fetch${channel?.name}`,
+          priority: QueueJobPriority.MEDIUM,
+          task: async () => {
+            user.saveOrUpdateIfExists(localDb);
+          }
+        });
+      } catch (e) {
+        console.log('error on save user from channel detail fetch', e);
+      }
+
+      return null;
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of a queue for saving channel data and user data is a good approach for managing asynchronous tasks. However, the error handling could be improved. If an error occurs during the execution of a task, it should be handled appropriately.
```diff
      queue.addPriorityJob({
        label: `saveChannelData-${channel?.id}`,
        priority: QueueJobPriority.MEDIUM,
        task: async () => {
          try {
            await channelData.saveIfLatest(localDb);
+            } catch (e) {
+              console.error(`Failed to save channel data ${channel?.id}`, e);
+            }
         }
       });
    }
 
    originalMembers?.map((member) => {
      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
      try {
        queue.addPriorityJob({
          label: `save user from channel detail fetch${channel?.name}`,
          priority: QueueJobPriority.MEDIUM,
          task: async () => {
            try {
              await user.saveOrUpdateIfExists(localDb);
+              } catch (e) {
+                console.error(`Failed to save or update user from channel detail fetch ${channel?.name}`, e);
+              }
          }
        });
      } catch (e) {
        console.log('error on save user from channel detail fetch', e);
      }

      return null;
    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->